werrors/errors.go

```diff
@@ -0,0 +1,85 @@
+package werrors
+
+import (
+	"fmt"
+)
+
+// New returns a new error instance. Avoid using Internal code when creating
+// errors.
+func New(cause Code, description string) error {
+	return &Error{
+		Code: cause,
+		Msg:  description,
+	}
+}
+
+// Wrap extends given error with additional information.
+//
+// If the wrapped error does not provide ABCICode method (ie. stdlib errors),
+// it will be labeled as internal error.
+func Wrap(err error, description string) error {
```

Comment:
what if we need to wrap and specify the code? Let's add a WrapWithCode method or something, otherwise looks good

Reply:
This was the detail I was trying to explain. I think that overwriting errors will cause issues and does not bring much value. 
Error codes no longer pin point to a single case. Instead, they represent classes of errors. for example  `NotFound` or `Unauthorized`. I cannot think of a real scenario where you want to keep the old error, but completely change it's meaning.

If you want to change the error, you can always use `New` to create a new one, for example:
```go
switch err := store.Get(accountKey) {
case nil:
    // all good
case werrors.NotFound.Is(err):
    return werrors.New(werrors.Unathorized, "invalid account")
default:
    return werrors.Wrap(err, "cannot fetch account")
}
```

Reply:
I had thought about this use case before as well but we should not wrap AND specify the code ever. As Piotr pointed out `New` would be the way to go for any error we want to handle with a specific code. If you for some reason have to preserve the original error message you can still pass it to `New`. This will give us a much cleaner api and also encourage either to handle the error or to pass it.

Reply:
Agreed